### PR TITLE
Fix large stack usage and add tests

### PR DESCRIPTION
**Pull request by swatanabe**
eos-vm change https://github.com/eosnetworkfoundation/mandel-eos-vm/pull/4 and unit tests

### CODE REVIEW

File: unittests/wasm_config_tests.cpp [hunk 1]
```diff
@@ -418,6 +418,75 @@ BOOST_DATA_TEST_CASE_F(old_wasm_tester, max_func_local_bytes_old, data::make({0,
    }
 }
 
+// Combines max_call_depth and max_func_local_bytes
+BOOST_FIXTURE_TEST_CASE(max_stack, wasm_config_tester) {
```
Comment (spoonincode) [inline unittests/wasm_config_tests.cpp:422]: The test name seems kind of a misnomer --  is it really testing the maximum? Or just something kinda large. My impression is that it's just kinda large.

Comment (swatanabe) [reply unittests/wasm_config_tests.cpp:422]: You're right.  It's one frame short of the max, since I didn't increase the stack usage of `apply`

Comment (spoonincode) [reply unittests/wasm_config_tests.cpp:422]: If we're going to test the max it'd be nice to see a max+1 test fail too.

Comment (swatanabe) [reply unittests/wasm_config_tests.cpp:422]: I don't think that a combined max+1 test adds anything over the separate max+1 tests for max_call_depth and max_func_local_bytes which already exist.

Comment (spoonincode) [reply unittests/wasm_config_tests.cpp:422]: yeah, gotcha. I believe I was thinking of a different test

**Comment by larryk85** [comment]
Please resolve merge conflicts.

**Comment by swatanabe** [comment]
> Please resolve merge conflicts.

I'm planning to make some changes to the test cases based on Matt's comments.  I just haven't gotten around to it yet.

**Comment by larryk85** [comment]
> 

Okay, that sounds good.  Just wanted to make sure you were aware.

**Review by larryk85 — APPROVED** [review]

**Pull request merged**